Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/function/AzureFunctionMvpModel.java

```diff
@@ -0,0 +1,161 @@
+        }
+        return app;
+    }
+
+    public void deleteFunction(String sid, String appId) throws IOException {
+        getFunctionAppsClient(sid).deleteById(appId);
+        subscriptionIdToFunctionApps.remove(sid);
+    }
+
+    public void restartFunction(String sid, String appId) throws IOException {
+        getFunctionAppsClient(sid).getById(appId).restart();
+    }
+
+    public void startFunction(String sid, String appId) throws IOException {
+        getFunctionAppsClient(sid).getById(appId).start();
+    }
+
+    public void stopFunction(String sid, String appId) throws IOException {
+        getFunctionAppsClient(sid).getById(appId).stop();
+    }
+
+    /**
+     * List app service plan by subscription id and resource group name.
+     */
+    public List<AppServicePlan> listAppServicePlanBySubscriptionIdAndResourceGroupName(String sid, String group)
+            throws IOException {
+        List<AppServicePlan> appServicePlans = new ArrayList<>();
+
+        Azure azure = AuthMethodManager.getInstance().getAzureClient(sid);
+        appServicePlans.addAll(azure.appServices().appServicePlans().listByResourceGroup(group));
+
+        return appServicePlans;
+    }
+
+    /**
+     * List app service plan by subscription id.
+     */
+    public List<AppServicePlan> listAppServicePlanBySubscriptionId(String sid) throws IOException {
+        return AuthMethodManager.getInstance().getAzureClient(sid).appServices().appServicePlans().list();
+    }
+
+    /**
+     * List all the Function Apps in selected subscriptions.
+     *
+     * @param forceReload flag indicating whether force to fetch latest data from server
+     * @return list of Function App
+     */
+    public List<ResourceEx<FunctionApp>> listAllFunctions(final boolean forceReload) {
+        final List<ResourceEx<FunctionApp>> functions = new ArrayList<>();
+        List<Subscription> subs = AzureMvpModel.getInstance().getSelectedSubscriptions();
+        if (subs.size() == 0) {
+            return functions;
+        }
+        Observable.from(subs).flatMap((sd) ->
+                Observable.create((subscriber) -> {
+                    try {
+                        List<ResourceEx<FunctionApp>> functionList = listFunctions(sd.subscriptionId(), forceReload);
+                        synchronized (functions) {
+                            functions.addAll(functionList);
+                        }
+                    } catch (IOException e) {
+                        // swallow exception and skip error subscription
+                    }
+                    subscriber.onCompleted();
+                }).subscribeOn(Schedulers.io()), subs.size()).subscribeOn(Schedulers.io()).toBlocking().subscribe();
+        return functions;
+    }
+
+    /**
+     * List all Function Apps by subscription id.
+     */
+    @NotNull
+    public List<ResourceEx<FunctionApp>> listFunctions(final String subscriptionId, final boolean forceReload)
```

Comment:
what's the diff between listFunctions listAllFunctions? seems name is confusing, we can expose only one of them public?

Reply:
listFunctions will list functions in specified subscription, rename to `listFunctionsInSubscription`